src/tokenizedSubaccounts/PPTSA.sol

```diff
@@ -0,0 +1,597 @@
+    PPTSAStorage memory $ = _getPPTSAStorage();
+    bool makerShouldBeSellingSpread = ($.isCallSpread == $.isLongSpread) ? isTaker : !isTaker;
+    if (makerShouldBeSellingSpread && strikeAmount <= 0) {
+      revert PPT_InvalidHighStrikeAmount();
+    } else if (!makerShouldBeSellingSpread && strikeAmount >= 0) {
+      revert PPT_InvalidHighStrikeAmount();
+    }
+  }
+
+  function _validateAndSortStrikes(IRfqModule.TradeData[] memory makerTrades)
+    internal
+    view
+    returns (StrikeData memory lowerStrike, StrikeData memory higherStrike)
+  {
+    if (makerTrades.length != 2) {
+      revert PPT_InvalidParams();
+    }
+    StrikeData memory strike1 = _validateAndCreateIndividualStrike(makerTrades[0]);
+    StrikeData memory strike2 = _validateAndCreateIndividualStrike(makerTrades[1]);
+
+    if (
+      makerTrades[0].asset != makerTrades[1].asset || makerTrades[0].asset != address(_getPPTSAStorage().optionAsset)
+        || strike1.expiry != strike2.expiry
+    ) {
+      revert PPT_InvalidParams();
+    }
+
+    if (strike1.strike > strike2.strike) {
+      return (strike2, strike1);
+    }
+    return (strike1, strike2);
+  }
+
+  function _validateAndCreateIndividualStrike(IRfqModule.TradeData memory trade)
+    internal
+    view
+    returns (StrikeData memory)
+  {
+    PPTSAStorage storage $ = _getPPTSAStorage();
+    (uint optionExpiry, uint optionStrike, bool isCall) = OptionEncoding.fromSubId(trade.subId.toUint96());
+
+    if (
+      optionExpiry < block.timestamp + $.ppParams.optionMinTimeToExpiry
+        || optionExpiry > block.timestamp + $.ppParams.optionMaxTimeToExpiry
+    ) {
+      revert PPT_OptionExpiryOutOfBounds();
+    }
+
+    (uint callPrice, uint putPrice,) = _getOptionPrice(optionExpiry, optionStrike);
+    uint markPrice = isCall ? callPrice : putPrice;
+    if ($.isCallSpread != isCall) {
+      revert PPT_WrongInputSpread();
+    }
+    return StrikeData({
+      strike: optionStrike,
+      expiry: optionExpiry,
+      markPrice: markPrice.toInt256(),
+      tradePrice: trade.price,
+      tradeAmount: trade.amount
+    });
+  }
+
+  function _verifyRfqExecute(StrikeData memory lowerStrike, StrikeData memory higherStrike, uint maxFee) internal view {
+    PPTSAStorage storage $ = _getPPTSAStorage();
+    uint strikeDiff = higherStrike.strike - lowerStrike.strike;
+    if (strikeDiff != _getPPTSAStorage().ppParams.strikeDiff) {
+      revert PPT_StrikePriceOutsideOfDiff();
+    }
+
+    // we always require the trade to be one sell and one buy of equal size, although prices can be different.
+    if (higherStrike.tradeAmount != -lowerStrike.tradeAmount) {
+      revert PPT_InvalidTradeAmount();
+    }
+
+    (uint openSpreads, uint baseBalance, int cashBalance) = _getSubAccountStats();
+
+    if (cashBalance < $.ppParams.maxNegCash) {
+      revert PPT_CannotTradeWithTooMuchNegativeCash();
+    }
+
+    _verifyRFQFee(maxFee, _getBasePrice());
```

Comment:
🔴 passing in base price when _verifyRFQFee takes in totalLegAmount

Reply:
Shoot. Should probably add a unit test for this. Thanks for calling this out.